frontend/projects/upgrade/src/app/core/experiments/experiment-design-stepper.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { selecthasExperimentStepperDataChanged } from './store/experiments.selectors';
+import { Store, select } from '@ngrx/store';
+import { AppState } from '../core.state';
+import * as experimentAction from './store//experiments.actions';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExperimentDesignStepperService {
+  expStepperDataChangedflag = false;
+  hasExperimentStepperDataChanged$: Observable<boolean>;
+
+  constructor(private store$: Store<AppState>) {
+    this.hasExperimentStepperDataChanged$ = this.store$.pipe(select(selecthasExperimentStepperDataChanged));
+    this.hasExperimentStepperDataChanged$.subscribe(
+      (isdataChanged) => (this.expStepperDataChangedflag = isdataChanged)
+    );
+  }
+
+  getHasExperimentDesignStepperDataChanged() {
+    return this.expStepperDataChangedflag;
+  }
+
+  experimentStepperDataChanged() {
+    this.store$.dispatch(experimentAction.experimentStepperDataChanged());
+  }
+
+  experimentStepperDataUpdated() {
```

Comment:
last request, can you change this name to `experimentStepperDataReset` so it matches the service call and is clear this is for reseting?

Reply:
I have made changes but I am facing this error in back-end testing while commit, I am facing same since I pulled code from dev today. Are you facing similar error? I have tried to pull from this branch but nothing helps so not sure what to do. 
![image](https://user-images.githubusercontent.com/50392803/202517397-70eb02e6-3fcf-4324-b7e3-2f841806ccd3.png)

Reply:
yes this looks like the Postgres db is not running, as the integration test suite requires a real db connection.

Reply:
If this continues to be an issue, I'm 100% these backend tests pass for this and I can see that the frontend tests passed, so in this case I'd say `--no-verify` is safe to commit and push this small change and I will merge the pr

Reply:
sorry I fall asleep later, I will ask someone from Playpower if they can merge this PR